servant-docs/src/Servant/Docs/Internal.hs

```diff
@@ -950,6 +955,27 @@ instance (KnownSymbol sym, ToParam (QueryFlag sym), HasDocs api)
           paramP = Proxy :: Proxy (QueryFlag sym)
           action' = over params (|> toParam paramP) action
 
+-- | The docs for a @'QueryParamForm' sym a'@
+--   require the following instances for the `a`:
+--   'Data', 'ToSample'
+instance (KnownSymbol sym, Data a, ToForm a, ToSample a, HasDocs api)
```

Comment:
`Data`, why?

Reply:
You're right. I can pull this out and we can use probably use the `ToSample` part with `urlEncodeAsForm` the sample supplied?